### ISSUE TITLE
Matching sources in different images during visit does not always work and creates duplicates in mergedMat

### ISSUE THREAD

**Issue by noralinn** [EranOfek/AstroPack#218]
The mergedMat file contains sometimes contains several sources with separations <3arcsec that are in reality the same source. This happens ~3 times per subimage. One third of the cases are bright, but not saturated, stars. The others are fainter sources.

Below an example for 
.../20/LAST.01.08.01_20230424.201701.955_clear_184-02_000_001_010_sci_merged_MergedMat_1.hdf5

or

.../20/LAST.01.08.01_20230424.201701.963_clear_184-02_001_001_010_sci_coadd_Image_1.fits

Coordinates and separation (arcsec) for badly matches sources:

183.75866104, -2.48868234, 1.40
183.75827905, -2.48859489, 1.40

183.6035654, -2.35164057, 1.66
183.60342382, -2.35208052, 1.66

183.94678214, -2.27483078, 1.72
183.94634944, -2.27462222, 1.72

Screenshots of coadded images:

![Screenshot from 2023-08-16 17-48-06](https://github.com/EranOfek/AstroPack/assets/14293969/ccb00f94-7bb4-49a6-b5f4-48be8d23c761)
![Screenshot from 2023-08-16 17-47-39](https://github.com/EranOfek/AstroPack/assets/14293969/92a4b21d-6b5a-4e83-bcf9-2
![Screenshot from 2023-08-16 18-05-38](https://github.com/EranOfek/AstroPack/assets/14293969/67b4b54f-010b-4aa1-80fe-f1122be9c73e)
d88a0634597)

**Comment by noralinn** [comment]
Every single image only contains a single detected source, so it isn't broken up, but the coordinates/astrometry between the images in the visit are not similar enough, so the problem happens when merging them to create the mergedMat file.

The bright stars is at the first position in 14 (out if 20 images) and at the second position (1.4 arcsec off) in the remaining 6 images.

For the two other sources it's: 
first position in 9 images and another one in 2 images, undetected in the remaining 9 images.
first position in 5 images, second in 12 images, undetected in 3 imgs.

**Comment by EranOfek** [comment]
are these stars saturated?

**Comment by noralinn** [comment]
Yes, the bright one is. I didn't realize before because it's a processed coadd, so I looked for the wrong number.

The two faint ones are close to the detection threshold, so far from saturated.

![Screenshot from 2023-08-18 13-16-37](https://github.com/EranOfek/AstroPack/assets/14293969/25da3cc5-6081-4f3c-9ee0-3b32938b68bb)

**Comment by EranOfek** [comment]
So this is in a single image.
So we will have to write the function I mentuoned earlier.
Thanks

**Comment by EranOfek** [comment]
mainly solved by multIterExtractor
